File: MyMoney.Core.FS/Models/BudgetItem.fs
namespace MyMoney.Core.FS.Models

type BudgetItem () =
    let mutable _Id = 0
    let mutable _Category = ""
    let mutable _Amount = Currency(0m)

    member this.Id
        with get () = _Id
        and set value = _Id <- value

    member this.Category
        with get () = _Category
        and set value = _Category <- value

    member this.Amount
        with get () = _Amount
        and set value = _Amount <- value

File: MyMoney.Core/Models/BudgetItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using CommunityToolkit.Mvvm.ComponentModel;

namespace MyMoney.Core.Models
{
    public partial class BudgetItem : ObservableObject, ICloneable
    {
        public object Clone()
        {
            return new BudgetItem()
            {
                Id = this.Id,
                Category = this.Category,
                Amount = new Currency(this.Amount.Value),
                Actual = new Currency(this.Actual.Value),
            };
        }

        [ObservableProperty]
        private int _id = 0;

        [ObservableProperty]
        private string _category = "";

        [ObservableProperty]
        private Currency _amount = new(0m);

        [ObservableProperty]
        private Currency _actual = new(0m);
    }
}
